shotover/src/transforms/kafka/sink_cluster/shotover_node.rs

```diff
@@ -0,0 +1,41 @@
+use crate::transforms::kafka::sink_cluster::kafka_node::KafkaAddress;
+use atomic_enum::atomic_enum;
+use kafka_protocol::messages::BrokerId;
+use kafka_protocol::protocol::StrBytes;
+use serde::{Deserialize, Serialize};
+use std::sync::Arc;
+
+#[derive(Serialize, Deserialize, Debug, Clone)]
+#[serde(deny_unknown_fields)]
+pub struct ShotoverNodeConfig {
+    pub address: String,
+    pub rack: String,
+    pub broker_id: i32,
+}
+
+impl ShotoverNodeConfig {
+    pub fn build(self) -> anyhow::Result<ShotoverNode> {
+        Ok(ShotoverNode {
+            address: KafkaAddress::from_str(&self.address)?,
+            rack: StrBytes::from_string(self.rack),
+            broker_id: BrokerId(self.broker_id),
+            state: Arc::new(AtomicShotoverNodeState::new(ShotoverNodeState::Up)),
+        })
+    }
+}
+
+#[derive(Clone)]
+pub struct ShotoverNode {
```

Comment:
we should be able to keep ShotoverNode and ShotoverNodeState private?

Reply:
Done!